common-instancetypes: Add deployment instructions (#760)
kubevirt/user-guide: docs/operations/deploy_common_instancetypes.md

# Deploy common-instancetypes

> The [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) provide a set of [instancetypes and preferences](../virtual_machines/instancetypes.md) to help create KubeVirt [`VirtualMachines`](http://kubevirt.io/api-reference/main/definitions.html#_v1alpha1_virtualmachine).

Reviewer: Why not v1 on the VM? https://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine

Author: Good question, that was copied and pasted.

> Beginning with the 1.1 release of KubeVirt, cluster-wide resources can be deployed directly through KubeVirt, without the need to install another operator.

Reviewer: You might need another sentence here to complete the thought, e.g.

"... cluster wide resources can be deployed directly... without another operator..."  This allows deployment of a set of default instance types and preferences along side KubeVirt.

> ## Enabling automatic deployment of common-instancetypes

Reviewer: ```suggestion
## Enable automatic deployment of common-instancetypes
```

It is a good habit to keep headings in the same tense.

> To enable the deployment of cluster-wide common-instancetypes through KubeVirt respectively `virt-operator`, the `CommonInstancetypesDeploymentGate` feature gate needs to be enabled.

Reviewer: Maybe "through the KubeVirt `virt-operator`"

See [Activating feature gates](activating_feature_gates.md) on how to enable it.

## Deploy common-instancetypes manually

For customization purposes or to install namespaced resources, common-instancetypes can also be deployed by hand.

To install all resources provided by the [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) project without further customizations, simply build with `kustomize` and apply:

```yaml

> $ kubectl kustomize https://github.com/kubevirt/common-instancetypes.git | kubectl apply -f -

Reviewer: ```suggestion
$ kubectl apply -k https://github.com/kubevirt/common-instancetypes.git
```

# Deploy common-instancetypes

The [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) provide a set of [instancetypes and preferences](../virtual_machines/instancetypes.md) to help create KubeVirt [`VirtualMachines`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine).

Beginning with the 1.1 release of KubeVirt, cluster wide resources can be deployed directly through KubeVirt, without another operator.
This allows deployment of a set of default instance types and preferences along side KubeVirt.

## Enable automatic deployment of common-instancetypes

To enable the deployment of cluster-wide common-instancetypes through the KubeVirt `virt-operator`, the `CommonInstancetypesDeploymentGate` feature gate needs to be enabled.

See [Activating feature gates](activating_feature_gates.md) on how to enable it.

## Deploy common-instancetypes manually

For customization purposes or to install namespaced resources, common-instancetypes can also be deployed by hand.

> To install all resources provided by the [`kubevirt/common-instancetypes`](https://github.com/kubevirt/common-instancetypes) project without further customizations, simply apply with `kustomize` enabled:

Reviewer 2: What does "`kustomize` enabled" mean? Could you please clarify?

```yaml
$ kubectl apply -k https://github.com/kubevirt/common-instancetypes.git
```

> Alternatively targets for each of the available custom resource types (e.g. namespaced instancetypes) are available.

Reviewer 2: ```suggestion
Alternatively, targets for each of the available custom resource types (e.g. namespaced instancetypes) are available.
```

Reviewer 2: ```suggestion
Alternatively, there are a variety of targets included for each of the custom resource types available. For instance, namespaced instance types (maybe here we can mention the specific target), name another one, etc.
```
